pkgs/development/python-modules/qiskit-ibmq-provider/default.nix

```
@@ -44,31 +56,16 @@ buildPythonPackage rec {
     requests
     requests_ntlm
     websockets
-    # Visualization/Jupyter inputs
-    ipykernel
-    ipyvuetify
-    ipywidgets
-    matplotlib
-    nbconvert
-    nbformat
-    plotly
-    pyperclip
-    seaborn
-  ];
-
-  # websockets seems to be pinned b/c in v8+ it drops py3.5 support. Not an issue here (usually py3.7+, and disabled for older py3.6)
-  postPatch = ''
-    substituteInPlace requirements.txt --replace "websockets>=7,<8" "websockets"
-    substituteInPlace setup.py --replace "websockets>=7,<8" "websockets"
-  '';
+  ] ++ lib.optional withVisualization visualizationPackages;
 
   # Most tests require credentials to run on IBMQ
   checkInputs = [
     pytestCheckHook
+    nbconvert
+    nbformat
     pproxy
     vcrpy
-  ];
-  dontUseSetuptoolsCheck = true;
+  ] ++ visualizationPackages;
```

this will be added twice to the check phase if withVisualization is passed
```suggestion
  ] ++ lib.optionals (!withVisualization) visualizationPackages;
```

---

Ok, I assumed there was some sort of unique filter when adding packages, but I have no objection.